force-app/main/default/aura/stratcraft/stratcraftHelper.js

```diff
@@ -1,158 +1,225 @@
-          }
-
-      );
-      //finally, update the node itself
-      //REFACTOR: rename this function to highlight expanded scope?
-      curNode.name = changedNode.name;
-      curNode.description = changedNode.description;
-      curNode.type = changedNode.type;
-      curNode.definition = changedNode.definition;
-      cmp.set("v.curStrat", curStrat);
+      tree.moveNode(curNode.parentNodeName, changedNode.parentNodeName, curNode.name);
+      //seems like we should also be adjusting the strategy here, and not just the tree
+    }
+  },
+
+  updateNodeName: function (cmp, curNode, changedNode) {
+    var self = this;
+
+    //first update the tree....
+    var tree = cmp.find('tree');
+    tree.renameNode(curNode, changedNode);
+
+    //then update the strategy model
+    //find any children of this node and update their parentNodeNames
+    var curStrat = cmp.get("v.curStrat");
+    var childNodes = self.findChildStrategyNodes(curStrat, curNode.name);
+    childNodes.forEach(function (child) {
+      child.parentNodeName = changedNode.name;
+    }
+
+    );
+    //finally, update the node itself
+    //REFACTOR: rename this function to highlight expanded scope?
+    curNode.name = changedNode.name;
+    curNode.description = changedNode.description;
+    curNode.type = changedNode.type;
+    curNode.definition = changedNode.definition;
+    cmp.set("v.curStrat", curStrat);
 
+  },
+
+  updateNodeParent: function (curNode, changedNode) {
+    curNode.parentNodeName = changedNode.parentNodeName;
+  },
+
+  saveStrategyChanges: function (cmp, changedNode, originalNodeName, helper) {
+
+    var curStrat = cmp.get("v.curStrat");
+    var curNode = helper.findStrategyNodeByName(curStrat, originalNodeName);
+
+    //if parent node was changed this is a move
+    if (curNode.parentNodeName !== changedNode.parentNodeName) {
+      helper.moveNode(cmp, curNode, changedNode);
+      helper.updateNodeParent(curNode, changedNode);
+    }
+
+    //if name was changed - also need to update nodes that are children of current node
+    if (curNode.name !== changedNode.name) {
+      helper.updateNodeName(cmp, curNode, changedNode);
+    }
+
+    cmp.set("v.curStrat", curStrat);
+  },
+
+  //this method checks if there are unsaved changes when user selects another node
+  //prompts user if he wants to save and saves, if so 
+  handleUnsavedChanged: function (component, newSelectedNodeName, curStrat, helper, continueSelectionCallback) {
+    var previousNodeName = component.find("propertyPage").get("v.originalName");
+
+    var needChecking = previousNodeName && newSelectedNodeName && (previousNodeName != newSelectedNodeName);
+    if (needChecking) {
+      var dirtyNode = component.find("propertyPage").get("v.curNode");
+      var originNode = helper.findStrategyNodeByName(curStrat, previousNodeName);
+      var originNodeName = originNode.name;
+      //possibly better to use underscore.js to compare 2 objects in a generic way
```

Comment:
one of the biggest weaknesses of the salesforce lightning/aura system is that it's hard to use external libraries. Not impossible. just something of a pain. They have to be registered as static resources. We can do it but should make sure it's worth it.